openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java

```diff
@@ -123,6 +137,74 @@ public Topic addSampleData(UUID topicId, TopicSampleData sampleData) throws IOEx
     return topic.withSampleData(sampleData);
   }
 
+  private void setFieldFQN(String parentFQN, List<Field> fields) {
+    fields.forEach(
+        c -> {
+          String fieldFqn = FullyQualifiedName.add(parentFQN, c.getName());
+          c.setFullyQualifiedName(fieldFqn);
+          if (c.getChildren() != null) {
+            setFieldFQN(fieldFqn, c.getChildren());
+          }
+        });
+  }
+
+  private void getFieldTags(boolean setTags, List<Field> fields) {
+    for (Field f : listOrEmpty(fields)) {
+      f.setTags(setTags ? getTags(f.getFullyQualifiedName()) : null);
+      getFieldTags(setTags, f.getChildren());
+    }
+  }
+
+  private void addDerivedFieldTags(List<Field> fields) {
+    if (nullOrEmpty(fields)) {
+      return;
+    }
+
+    for (Field field : fields) {
+      field.setTags(addDerivedTags(field.getTags()));
+      if (field.getChildren() != null) {
+        addDerivedFieldTags(field.getChildren());
+      }
+    }
+  }
+
+  List<Field> cloneWithoutTags(List<Field> fields) {
+    if (nullOrEmpty(fields)) {
+      return fields;
+    }
+    List<Field> copy = new ArrayList<>();
+    fields.forEach(f -> copy.add(cloneWithoutTags(f)));
+    return copy;
+  }
+
+  private Field cloneWithoutTags(Field field) {
+    List<Field> children = cloneWithoutTags(field.getChildren());
+    return new Field()
+        .withDescription(field.getDescription())
+        .withName(field.getName())
+        .withDisplayName(field.getDisplayName())
+        .withFullyQualifiedName(field.getFullyQualifiedName())
+        .withDataType(field.getDataType())
+        .withChildren(children);
+  }
+
+  private void applyTags(List<Field> fields) {
+    // Add column level tags by adding tag to column relationship
+    for (Field field : fields) {
+      applyTags(field.getTags(), field.getFullyQualifiedName());
+      if (field.getChildren() != null) {
+        applyTags(field.getChildren());
+      }
+    }
+  }
+
+  @Override
+  public void applyTags(Topic topic) {
+    // Add table level tags by adding tag to table relationship
+    super.applyTags(topic);
+    applyTags(topic.getSchemaFields());
+  }
+
   public class TopicUpdater extends EntityUpdater {
     public TopicUpdater(Topic original, Topic updated, Operation operation) {
```

Comment:
TopicUpdater may need changes to record field changes just like we do for table columns. This can be done in another issue.